frontend/apps/web/components/jobs/NosqlTable/NosqlTable.tsx

```diff
@@ -253,7 +349,26 @@ function AddNewRecord(props: AddNewRecordProps): ReactElement {
                 The collection that you want to map.
               </FormDescription>
               <FormControl>
-                <Select onValueChange={field.onChange} value={field.value}>
+                <Select
+                  onValueChange={(value) => {
+                    field.onChange(value);
+                    form.clearErrors('key');
+                    const currentKey = form.getValues('key');
+                    if (currentKey) {
+                      const lastDotIndex = value.lastIndexOf('.');
+                      const schema = value.substring(0, lastDotIndex);
+                      const table = value.substring(lastDotIndex + 1);
+                      if (isDuplicateKey(currentKey, schema, table)) {
+                        form.setError('key', {
+                          type: 'manual',
+                          message:
+                            'This key already exists in the selected collection.',
+                        });
+                      }
```

Comment:
This is used to catch the scenario where the user selects the collection in the new mapping form`a` and then enters  key 'b'. It returns back a `true` for 'isDuplicateKey()'. Then they try to change the collection to `b`. If collection `b` already has key 'a', then we need to throw the `isDuplicateKey()` error here. But since the uniqueMapping sits on the `key` field, it doesn't do that since the key field isn't changing, it's the collection field. 

I suppose we could move this into a `.(test)` function on the collection. I haven't tried it and my initial thought was to do it on the select, so it might work. Would need to test it.